diff --git a/app/assets/javascripts/discourse/components/topic-navigation.js.es6 b/app/assets/javascripts/discourse/components/topic-navigation.js.es6
index ecae615..4af6b6c 100644
--- a/app/assets/javascripts/discourse/components/topic-navigation.js.es6
+++ b/app/assets/javascripts/discourse/components/topic-navigation.js.es6
@@ -105,6 +105,7 @@ export default Ember.Component.extend({
   },
 
   didInsertElement() {
+    var exec = () => {
     this._super();
 
     this.appEvents
@@ -120,6 +121,11 @@ export default Ember.Component.extend({
     }
 
     this._checkSize();
+
+    };
+
+    if(!$('html').hasClass('game-discuss')) exec();
+    else setTimeout(exec, 2500);
   },
 
   willDestroyElement() {
