diff --git a/app/assets/javascripts/discourse/components/topic-navigation.js.es6 b/app/assets/javascripts/discourse/components/topic-navigation.js.es6
index 1f06f87..b6ef54d 100644
--- a/app/assets/javascripts/discourse/components/topic-navigation.js.es6
+++ b/app/assets/javascripts/discourse/components/topic-navigation.js.es6
@@ -110,6 +110,7 @@ export default Ember.Component.extend({
   },
 
   didInsertElement() {
+    var exec = () => {
     this._super();
 
     this.appEvents
@@ -125,6 +126,11 @@ export default Ember.Component.extend({
     }
 
     this._checkSize();
+
+    };
+
+    if(!$('html').hasClass('game-discuss')) exec();
+    else setTimeout(exec, 2500);
   },
 
   willDestroyElement() {
