--- a/app/assets/javascripts/discourse/components/topic-navigation.js.es6	2016-10-28 10:22:24.143613084 +0000
+++ b/app/assets/javascripts/discourse/components/topic-navigation.js.es6	2016-10-28 10:23:27.411613430 +0000
@@ -96,6 +96,7 @@
   },
 
   didInsertElement() {
+    var exec = () => {
     this._super();
 
     this.appEvents
@@ -111,6 +112,11 @@
     }
 
     this._checkSize();
+
+    };
+
+    if(!$('html').hasClass('game-discuss')) exec();
+    else setTimeout(exec, 2500);
   },
 
   willDestroyElement() {
