diff --git a/config/routes.rb b/config/routes.rb
index 36c265d..bae0fec 100644
--- a/config/routes.rb
+++ b/config/routes.rb
@@ -292,6 +292,7 @@ Discourse::Application.routes.draw do
   get "session/sso_provider" => "session#sso_provider"
   get "session/current" => "session#current"
   get "session/csrf" => "session#csrf"
+  get "session/logout" => "session#destroy"
   get "composer_messages" => "composer_messages#index"
 
   resources :static
